lit_tests/kernel/wave/attention/decode_attention.py

```diff
@@ -63,24 +63,25 @@ def test_paged_flash_decoding():
     ):
         print(phase_0(q, k, v, logits, logits_max))
 
-    # CHECK:                func.func @phase_0
+    # CHECK-LABEL:          func.func @phase_0
     # CHECK-DAG:               %[[C0:.*]] = arith.constant 0 : index
     # CHECK-DAG:               %[[C1:.*]] = arith.constant 1 : index
-    # CHECK-COUNT-4:           vector.load
-    # CHECK:                   scf.for %{{.*}} = %[[C0]] to %[[COUNT:.*]] step %[[C1]]
+    # CHECK-DAG:               %[[C32:.*]] = arith.constant 32 : index
+    # CHECK-COUNT-2:           vector.load
+    # CHECK:                   %[[COUNT0:.*]] = arith.minsi %{{.*}}, %{{.*}} : vector<1xindex>
+    # CHECK:                   %[[COUNT1:.*]] = vector.extract %[[COUNT0]][0] : index from vector<1xindex>
+    # CHECK-COUNT-2:           vector.load
+    # CHECK:                   %[[D36:.*]] = arith.subi %[[COUNT1]], %[[C1]] : index
+    # CHECK:                   %[[D37:.*]] = arith.divui %[[D36]], %[[C32]] : index
+    # CHECK:                   %[[D38:.*]] = arith.addi %[[D37]], %[[C1]] : index
+    # CHECK:                   %[[COUNT2:.*]] = arith.select %{{.*}}, %[[C0]], %38 : index
+    # CHECK:                   scf.for %{{.*}} = %[[C0]] to %[[COUNT2]] step %[[C1]]
     # CHECK:                        amdgpu.lds_barrier
-    # 1 masked load for sequence idx, 4 for k_cache, and 4 for v_cache.
-    # CHECK-COUNT-1:                vector.maskedload
-    # CHECK-NEXT:                   arith.index_cast
-    # CHECK-COUNT-4:                vector.maskedload
-    # CHECK-NEXT:                   vector.store
-    # CHECK-COUNT-4:                vector.maskedload
-    # CHECK-NEXT:                   vector.store
+    # 2 masked load block table, 2 for k_cache, and 2 for v_cache.
```

Comment:
Are these the additional loads introduced in global_to_shared_gathers.py?

Reply:
Not by `global_to_shared_gathers` specifically, but by just paged decode rewrite